### PR TITLE
Soft poweroff 2

### PR DESCRIPTION
**Pull request by Xenwar**

### CODE REVIEW

**Comment by nordixinfra** [comment]
Can one of the admins verify this patch?

File: pkg/controller/baremetalhost/baremetalhost_controller.go [hunk 1]
```diff
@@ -726,8 +726,19 @@ func (r *ReconcileBareMetalHost) manageHostPower(prov provisioner.Provisioner, i
...
+		}
+
 	} else {
-		provResult, err = prov.PowerOff()
+		provResult, err = prov.SoftPowerOff()
+		if err = r.markLastSoftPowerOff(info.host); err != nil {
+			return reconcile.Result{}, errors.Wrap(err, fmt.Sprintf("failed to mark last soft power off time %q", err.Error()))
+		}
+		if r.softPowerOffTimedOut(info.host) {
+			provResult, err = prov.SoftPowerOff()
```
Comment (maelk) [inline pkg/controller/baremetalhost/baremetalhost_controller.go:735]: should be hard poweroff

File: pkg/controller/baremetalhost/baremetalhost_controller.go [hunk 2]
```diff
@@ -726,8 +726,19 @@ func (r *ReconcileBareMetalHost) manageHostPower(prov provisioner.Provisioner, i
 
 	if info.host.Spec.Online {
 		provResult, err = prov.PowerOn()
+		if err = r.unMarkLastSoftPowerOff(info.host); err != nil {
+			return reconcile.Result{}, errors.Wrap(err, fmt.Sprintf("failed to mark last soft power off time %q", err.Error()))
+		}
+
 	} else {
-		provResult, err = prov.PowerOff()
+		provResult, err = prov.SoftPowerOff()
```
Comment (maelk) [inline pkg/controller/baremetalhost/baremetalhost_controller.go]: Avoid sending multiple softpoweroff when it has already been sent

**Review by maelk — CHANGES_REQUESTED** [review]

**Review by Jaakko-Os — COMMENTED** [review]
looks good to me

**Comment by jan-est** [comment]
looks good to me

File: pkg/controller/baremetalhost/baremetalhost_controller.go (hunk 1 above)
Comment (Xenwar) [reply pkg/controller/baremetalhost/baremetalhost_controller.go:735]: done

File: pkg/controller/baremetalhost/baremetalhost_controller.go (hunk 2 above)
Comment (Xenwar) [reply pkg/controller/baremetalhost/baremetalhost_controller.go]: done

**Review by Jaakko-Os — APPROVED** [review]

**Pull request closed without merging**